fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1951,34 +1999,56 @@ public List<QueryComponent> combineNonSargables() {
 
         @Nonnull
         public ScoredPlan withPlan(@Nonnull RecordQueryPlan newPlan) {
-            return new ScoredPlan(newPlan, unsatisfiedFilters, indexFilters, score, createsDuplicates, includedRankComparisons);
+            return new ScoredPlan(newPlan, unsatisfiedFilters, indexFilters, sargedComparisons, score, createsDuplicates, includedRankComparisons);
         }
 
         @Nonnull
         public ScoredPlan withScore(int newScore) {
             if (newScore == score) {
                 return this;
             } else {
-                return new ScoredPlan(plan, unsatisfiedFilters, indexFilters, newScore, createsDuplicates, includedRankComparisons);
+                return new ScoredPlan(plan, unsatisfiedFilters, indexFilters, sargedComparisons, newScore, createsDuplicates, includedRankComparisons);
             }
         }
 
+        public Set<String> getSargedInBindings() {
+            return sargedInBindingsSupplier.get();
+        }
+
+        private Set<String> computeSargedInBindings() {
+            return sargedComparisons.stream()
+                    .filter(comparison -> comparison instanceof Comparisons.ParameterComparison)
+                    .map(comparison -> (Comparisons.ParameterComparison)comparison)
+                    .filter(parameterComparison -> Bindings.Internal.IN.isOfType(parameterComparison.getParameter()))
```

Comment:
Maybe it's too much trouble, but there's part of me that thinks that rather than using the name of the parameter here to compute the number of sarged IN-bindings, the list of comparisons in the `IN` extractor could be compared to the sarged comparisons. The current approach has the unfortunate side-effect that if a user creates a parameter called `__in_blah_blah`, it will be treated as an IN-parameter by this code. I'm not actually sure if we ban such parameters or not

Reply:
We do ban such parameters.